subsys/pm/device.c

```diff
@@ -133,6 +140,36 @@ int pm_device_action_run(const struct device *dev,
 	return 0;
 }
 
+void pm_device_children_action_run(const struct device *dev,
+		enum pm_device_action action,
+		pm_device_action_failed_cb_t failure_cb)
+{
+	const device_handle_t *handles;
+	size_t handle_count = 0;
+	int rc = 0;
+
+	/*
+	 * We don't use device_supported_foreach here because we don't want the
+	 * early exit behaviour of that function. Even if the N'th device fails
+	 * to PM_DEVICE_ACTION_TURN_ON for example, we still want to run the
+	 * action on the N+1'th device.
+	 */
+	handles = device_supported_handles_get(dev, &handle_count);
+
+	for (size_t i = 0; i < handle_count; ++i) {
+		device_handle_t dh = handles[i];
+		const struct device *cdev = device_from_handle(dh);
+
+		rc = pm_device_action_run(cdev, action);
+		if (failure_cb && rc < 0) {
```

Comment:
maybe `-ENOTSUP` should require special handling (if action if optional should not be treated as an error)

Reply:
I would prefer to leave special handling to the callback, maybe someone wants a strict regulator that warns if you put a sensor on it that doesn't handle power up/down